CMSgov/design-system
# [WNMGDS-545] Fix rootPath option

### Summary
The `rootPath` option is needed in order to update HTML build files to have the correct paths when the doc site is published. For example, if the doc site is published at `domain.com/design/`, the `rootPath` would be set to `design`, and the `index.html` will link to `href="/design/index.css` instead of `href="/index.css"`.

The way `rootPath` used to work was the HTML paths would be updated **and** a [duplicate set of build files](https://github.cms.gov/CMS-WDS/ds-healthcare-gov/tree/master/docs) would be generated inside directories that match the root path. The duplicate build files are used to serve the site locally (i.e. `localhost:3000/pages/CMS-WDS/ds-healthcare-gov`) while the build files at the root (i.e. files under `docs/`) are the files that are actually used on the Github Pages site.

In this PR, when `rootPath` is set in the cmsds config file **and** you are building for production, the resulting build files will use the `rootPath` in their paths. The `rootPath` is ignored in local development so that the files served by `browsersync` will work correctly. You can also ignore `rootPath` when running `yarn build` by providing the `ignoreRootPath` option, this is used when you want to build in production but also want to serve the site locally, like when running e2e tests.

### How to test
- Run `yarn start` and quickly ensure everything is normal
- Go to `cli.js` and comment out line 89 under `yarn start`, and re run the command
- You then need to move all the files inside `docs/dist` to `docs/dist/design-system`, and then go to `localhost:3000/design-system/`. This simulates publishing to a domain with `design-system` as the `rootPath`.
- Test the entire site, especially all links to other parts of the site
- Undo all those changes and run `yarn test:e2e`
